src/attr.c

```diff
@@ -333,6 +333,7 @@ static int attr_setup(git_repository *repo, git_attr_session *attr_session)
 				       NULL, git_repository_attr_cache(repo)->cfg_attr_file)) < 0)
 		goto out;
 
+	git_buf_clear(&path); /* git_repository_item_path expects an empty buffer, becasue it uses git_buf_set */
```

Comment:
Yup, this fix definitely looks correct to me. Previously, we would've usually never encountered this code path due to the wrong early exit.

Anyway. I'll have to investigate why we have so much duplicated code in `attr_setup` and `collect_attr_files`.

Reply:
First of all I'd like to have this regression fixed.

Reply:
Cannot argue with that. Does your proposed change fix the issues you're seeing? If so, then we may just merge this quickly and be done with it. I'm currently writing a testcase that I'd like to include here, shouldn't take me too long.

Anyway, why we have two passes: we need to preload gitattributes files as they may contain macros that are used in other files. So the first pass is only to make macros available, the second pass is to re-parse files to make use of macros. But in fact, we do this the wrong way. gitattributes(5) specifies that macros may only be parsed from either global files, the repo's file in ".git/info/attributes" or the top-level working directory only. It explicitly says that macros defined in subdirectories shall _not_ be honored, but we fail to respect this. I'll create a PR to fix this.